param/parameterized.py

```diff
@@ -981,6 +982,13 @@ def __getattribute__(mcs,name):
             return type.__getattribute__(mcs,name)
 
 
+@lru_cache()
+def _update_signature():
```

Comment:
I can see this is the same name as used by the class. Open to better suggestions for naming.

Reply:
This could be a constant as this will be called during initialization.